controllers/usersignup/approval_test.go

```diff
@@ -60,6 +59,60 @@ func TestGetClusterIfApproved(t *testing.T) {
 		assert.Equal(t, "member2", clusterName.getClusterName())
 	})
 
+	t.Run("with two clusters available, the second one has required cluster-role label", func(t *testing.T) {
+		// given
+		signup := commonsignup.NewUserSignup()
+		toolchainConfig := commonconfig.NewToolchainConfigObjWithReset(t,
+			testconfig.AutomaticApproval().
+				Enabled(true),
+			testconfig.CapacityThresholds().
+				MaxNumberOfSpaces(testconfig.PerMemberCluster("member1", 1000), testconfig.PerMemberCluster("member2", 1000)).
+				ResourceCapacityThreshold(80, testconfig.PerMemberCluster("member1", 70), testconfig.PerMemberCluster("member2", 75)))
+		fakeClient := NewFakeClient(t, toolchainStatus, toolchainConfig)
+		InitializeCounters(t, toolchainStatus)
+		clusters := NewGetMemberClusters(
+			NewMemberClusterWithoutClusterRoles(t, "member1", corev1.ConditionTrue), // no cluster-role label on this member
+			NewMemberCluster(t, "member2", corev1.ConditionTrue),                    // by default all member clusters will have the 'tenant' cluster-role
+		)
+
+		// when
+		approved, clusterName, err := getClusterIfApproved(fakeClient, signup, capacity.NewClusterManager(clusters, fakeClient))
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, approved)
+		assert.Equal(t, "member2", clusterName.getClusterName())
+	})
+
+	t.Run("return preferred cluster name even if without required cluster role", func(t *testing.T) {
+		// given
+		signup := commonsignup.NewUserSignup()
+		signup.Annotations = map[string]string{
+			toolchainv1alpha1.UserSignupLastTargetClusterAnnotationKey: "member1", // set preferred member cluster name
+		}
+		toolchainConfig := commonconfig.NewToolchainConfigObjWithReset(t,
+			testconfig.AutomaticApproval().
+				Enabled(true),
+			testconfig.CapacityThresholds().
+				MaxNumberOfSpaces(testconfig.PerMemberCluster("member1", 1000), testconfig.PerMemberCluster("member2", 1000)).
+				ResourceCapacityThreshold(80, testconfig.PerMemberCluster("member1", 70), testconfig.PerMemberCluster("member2", 75)))
+		fakeClient := NewFakeClient(t, toolchainStatus, toolchainConfig)
+		InitializeCounters(t, toolchainStatus)
+		clusters := NewGetMemberClusters(
+			// member1 doesn't have the cluster-role tenant but it's preferred one
+			NewMemberClusterWithoutClusterRoles(t, "member1", corev1.ConditionTrue),
+			NewMemberCluster(t, "member1", corev1.ConditionTrue), // by default all member clusters will have the 'tenant' cluster-role
```

Comment:
you already created member1 in the previous line
```suggestion
			NewMemberCluster(t, "member2", corev1.ConditionTrue), // by default all member clusters will have the 'tenant' cluster-role
```

Reply:
thanks for spotting this , fixed in [6b01926](https://github.com/codeready-toolchain/host-operator/pull/736/commits/6b01926137a7f5c783ed0b886e61eaba8cc13642)